controllers/AuthController.js

```diff
@@ -138,4 +138,48 @@ export default class AuthController {
       return next(err);
     }
   }
+
+  /**
+   *
+   * @static
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @param {function} next
+   * @returns {object} returns user data
+   */
+  static async socialLogin(req, res, next) {
+    try {
+      if (!req.user) {
+        return ServerResponse.notFoundError(req, res);
+      }
+      // eslint-disable-next-line no-underscore-dangle
+      const userData = req.user._json;
+      const createdUser = await User.findOrCreate({
+        where: { email: userData.email },
+        defaults: {
+          firstName: userData.name.split(' ')[0],
+          lastName: userData.name.split(' ')[1],
+          email: userData.email,
+          isVerified: true,
+          password: 'notrequired',
+        }
+      });
+      const token = Helper.createToken({
+        id: createdUser.id,
+        email: userData.email
+      });
+      return res.status(201).json({
+        user: {
+          token,
+          id: createdUser[0].id,
+          firstName: createdUser[0].firstName,
+          lastName: createdUser[0].lastName,
+          email: createdUser[0].email,
+          isVerified: createdUser[0].isVerified,
+        },
```

Comment:
One last change and you are good to go. Kindly destructure these `createdUser` properties

Reply:
Ok.. working on it!

Reply:
done